tcl/util.tcl

```diff
@@ -708,4 +750,133 @@ proc qc::regexp_escape {string} {
     } 
 
     return [string map $list $string]
+ }
+
+proc qc::split2 {args} {
+    #| Split a string into a list, ignoring quoted or escaped delimiters
+    qc::args $args -escape \\ -quote \" -- string delimiter
+    set length [string length $string]
+    set list {}
+    
+    # Loop with safety limit
+    for {set i $length} {$i>0} {incr i -1} {
+        set pair [qc::split_pair2 \
+                      -nocomplain \
+                      -escape $escape \
+                      -quote $quote \
+                      $string $delimiter]
+        if { [llength $pair] == 1 } {
+            lappend list [lindex $pair 0]
+            break
+        } else {
+            lappend list [lindex $pair 0]
+            set string [lindex $pair 1]
+        }
+    }
+    return $list
+}
+
+proc qc::split_pair2 {args} {
+    #| Split a string into 2 parts at the first occurence of the delimiter,
+    #| ignoring quoted or escaped delimiters
+    qc::args $args -nocomplain -escape \\ -quote \" -- string delimiter
+    default nocomplain false
```

Comment:
Rather than complain would it be possible to test the condition of the string first to anticipate if it can be split or not.

Reply:
It would be possible, but the code for testing would have roughly the same complexity as the code for splitting, and there would be a lot of duplication between the two. Overall I'd say it would increase the complexity of the code we'd need to maintain going forward.